internal/internal_task_handlers.go

```diff
@@ -891,6 +910,21 @@ ProcessEvents:
 				return nil, err
 			}
 
+			// TODO: update history size here, when isLast (implies decision-task started event is current, may execute new code)
+			if isLast {
+				// finished processing history prior to a decision -> update for that decision's execution
+				// because it may depend on the value
+				w.workflowInfo.HistorySize += historySize
+				/*
+					TODO: can't expose this unless the history is currently executing on a "new library version" that is history-size aware,
+						else you can get inconsistent replay behavior when it suddenly jumps from 0 (nonexistent data in the past) to N (calculated now during replay).
+						so this needs a new piece of metadata in decision task complete (possibly activity would be useful too for troubleshooting?) that reports the
+						library "feature version" for backwards compat reasons like this.
+
+					alternatively: could just record a new marker to say "upgraded to version N", but that means a new marker in every workflow.  seems like wasteful storage?
+				*/
```

Comment:
correction: no, not necessary.
any code that's added that changes behavior based on this new data needs to be `GetVersion`ed anyway, so that should take care of determinism based on this info existing vs not existing.